### PR TITLE
[RN-206] Configure logging

### PR DESCRIPTION
**Pull request by LCaparelli**
Changelog:

  - control logging behavior through env vars
  - add example .env
  - document env vars
  - decorate controller log with current reconciliation request namespaced name

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -47,6 +51,7 @@ func init() {
 	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
 
 	//+kubebuilder:scaffold:scheme
+	godotenv.Load()
```
Comment (caiofralmeida) [inline main.go:54]: It could be in `init` func? what do you think?

File: main.go [hunk 2]
```diff
@@ -58,13 +63,17 @@ func main() {
...
 	opts.BindFlags(flag.CommandLine)
 	flag.Parse()
 
-	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
+	operatorCfg := config.Parse()
+	
+	ctrl.SetLogger(zap.New(
+		zap.UseFlagOptions(&opts),
+		zap.UseDevMode(operatorCfg.DevMode),
+		zap.Level(mustGetLogLevel(operatorCfg.LogLevel)),
```
Comment (caiofralmeida) [inline main.go:75]: Another option could be avoided is to have a panic-killing app with status code 1. What do you think

**Review by caiofralmeida — APPROVED** [review]

File: main.go (hunk 1 above)
Comment (LCaparelli) [reply main.go:54]: But it is in the `init` func, I'm confused hehe

File: main.go (hunk 2 above)
Comment (LCaparelli) [reply main.go:75]: A panic exits with non-zero:

```
❯ cat /tmp/panic.go
package main

func main() {
    panic("oh no")
}
❯ go run /tmp/panic.go
panic: oh no

goroutine 1 [running]:
main.main()
	/tmp/panic.go:4 +0x39
exit status 2
```
Do you think we should push specifically for status code 1?

File: main.go (hunk 1 above)
Comment (caiofralmeida) [reply main.go:54]: The github code review tool troll me heheh

File: main.go (hunk 2 above)
Comment (caiofralmeida) [reply main.go:75]: As we talked, the panic returns the message feedback for user, so I consider ok.

**Pull request merged**